proposals/csharp-9.0/records.md

```diff
@@ -197,6 +197,118 @@ If the containing record is abstract, the synthesized clone method is also abstr
 If the "clone" method is not abstract, it returns the result of a call to a copy constructor. 
 
 
+### Printing members: PrintMembers and ToString
+
+If the record is derived from `object`, the record includes a synthesized method equivalent to a method declared as follows:
+```C#
+protected virtual void PrintMembers(System.StringBuilder builder, bool includeSeparator);
+```
+The method is `virtual` and `protected`.
+It is an error if the method is declared explicitly.
+
+The method:
+1. appends a separator ", " to `builder` if `includeSeparator` is true and the record has public fields or properties,
+2. for each of the record's public field and property members, appends that member's name followed by " = " followed by the result of invoking `object.ToString()` on that member's value: `this.member.ToString()`, separated with ", ",
```

Comment:
is this more like objectdisplay?, or more like a jsut a sequence of tostring'ed members?  for example, if i have a string-member, will this be printed with quotes?

Reply:
string members will not be printed with quotes.
See behavior of ToString on anonymous types ([sharplab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxAgrgOwD4AEBMAjALABQuAzAAQFUDCVA3mVazfvky2zwG4QIqADyoBeKpjgB3JlQCCYqgCIAFnAA26gPZKqAXwDc3Hq1yEAdHS2YoW9XHMB1BAEt4ACiHmAKloDKMK6YAObuAJRhRqQ8emR6QA))

Reply:
THat seems like a really bad default (Esp. as we can't change this in teh future).  i get it for a debugger display, but the actual ToString of these types, it seems very insufficient and actually actively unhelpful for anyone who might use the ToString here.

Reply:
@CyrusNajmabadi But then you have to handle escaping the quotes or the you've only kicked the can down the road

Reply:
> @CyrusNajmabadi But then you have to handle escaping the quotes or then you've only kicked the can down the road

Yes :)  That's what 'go' does, what the debugger does, etc. etc.  taht seems like the sane approach to me if this is part of the public contract here.

Reply:
@CyrusNajmabadi Aside from anonymous types, we also have the precedent of tuples which do not quote or escape ([sharplab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxAgrgOwD4AEBMAjALABQuAzAAQFUDCVA3mVazfvky2z7oQHR0A9pihCANnH4B1BAEt4ACkUAiABZxx4oSqRUALPgCURgNzdWAXzKWgA===)).
I'll stick with that precedent.